app/services/opd_service/reports/drug.rb

```diff
@@ -0,0 +1,145 @@
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%qnoon%' THEN i.dose * 1 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%qpm%' THEN i.dose * 1 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%qhs%' THEN i.dose * 1 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%qod%' THEN i.dose * 0.5 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%qwk%' THEN i.dose * 0.14 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%once a month%' THEN i.dose * 0.03 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%twice a month%' THEN i.dose * 0.071 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        WHEN LOWER(i.frequency) LIKE '%unknown%' THEN i.dose * 0 * (
+          CASE 
+            WHEN LOWER(o.instructions) LIKE '% for %' 
+              THEN SUBSTRING(o.instructions, LOCATE('for', o.instructions) + 4)
+            ELSE 1
+          END
+        )
+        ELSE i.dose 
+      END AS prescribe_quantity,(
+        SELECT GROUP_CONCAT(c.name SEPARATOR ', ') AS names FROM encounter e 
+        INNER JOIN obs ON obs.encounter_id = e.encounter_id
+        INNER JOIN concept_name c ON c.concept_id = obs.value_coded 
+        WHERE e.`voided` = 0 AND (DATE(encounter_datetime) BETWEEN '#{start_date}' AND '#{end_date}'
+        AND encounter_type = 8 -- OUTPATIENT DIAGNOSIS
+        AND obs.person_id = encounter.patient_id
+        AND obs.concept_id IN(6543 -- Secondary diagnosis
+          ,6542 -- Primary diagnosis
+          )) 
+        AND Date(e.date_created) = DATE(o.date_created)
+      ) as diagnosis,
```

Comment:
This will definitely make the code slower. Best to have simple select in the select sections. I believe a restructured query will be ideal